sims/xcelium/Makefile

```diff
@@ -0,0 +1,138 @@
+sim_dir=$(abspath .)
+
+#########################################################################################
+# include shared variables
+#########################################################################################
+include $(base_dir)/variables.mk
+
+#########################################################################################
+# name of simulator (used to generate *.f arguments file)
+#########################################################################################
+sim_name = xrun
+
+#########################################################################################
+# xcelium simulator types and rules
+#########################################################################################
+sim_prefix = simx
+sim = $(sim_dir)/$(sim_prefix)-$(MODEL_PACKAGE)-$(CONFIG)
+sim_debug = $(sim)-debug
+sim_workdir = $(sim).xcelium.d
+sim_run_tcl = $(sim)_run.tcl
+sim_debug_run_tcl = $(sim_debug)_run.tcl
+
+include $(base_dir)/xcelium.mk
+
+.PHONY: default debug
+default: $(sim)
+debug: $(sim_debug)
+
+#########################################################################################
+# simulation requirements
+#########################################################################################
+SIM_FILE_REQS += \
+	$(ROCKETCHIP_RSRCS_DIR)/vsrc/TestDriver.v
+
+# copy files but ignore *.h files in *.f since xcelium has -Wcxx include
+$(sim_files): $(SIM_FILE_REQS) $(ALL_MODS_FILELIST) | $(GEN_COLLATERAL_DIR)
+	cp -f $(SIM_FILE_REQS) $(GEN_COLLATERAL_DIR)
+	$(foreach file,\
+		$(SIM_FILE_REQS),\
+		$(if $(filter %.h,$(file)),\
+			,\
+			echo "$(addprefix $(GEN_COLLATERAL_DIR)/, $(notdir $(file)))" >> $@;))
+
+#########################################################################################
+# import other necessary rules and variables
+#########################################################################################
+include $(base_dir)/common.mk
+
+#########################################################################################
+# xcelium binary and arguments
+#########################################################################################
+XCELIUM = xrun
+XCELIUM_OPTS = $(XCELIUM_CC_OPTS) $(XCELIUM_NONCC_OPTS) $(PREPROC_DEFINES)
+
+#########################################################################################
+# xcelium build paths
+#########################################################################################
+model_dir = $(build_dir)/$(long_name)
+model_dir_debug = $(build_dir)/$(long_name).debug
+
+
+#########################################################################################
+# xcelium simulator rules
+#########################################################################################
+
+$(sim_workdir): $(sim_common_files) $(dramsim_lib) $(EXTRA_SIM_REQS)
+	rm -rf $(model_dir)
+	$(XCELIUM) -elaborate $(XCELIUM_OPTS) $(EXTRA_SIM_SOURCES) $(XCELIUM_COMMON_ARGS)
+
+$(sim_run_tcl): $(sim_workdir)
```

Comment:
Can you somehow inline the tcl so that you don't create an extra run tcl script?

Reply:
Track this issue in https://github.com/ucb-bar/chipyard/issues/1405